feat: jwt를 활용한 CUD 작업

leehyunk6310 opened the pull request:
- 토큰 시간 100분으로 수정
- dto에 time 추가
- 엔티티의 link -> url 변경

leehyunk6310 committed:
feat: jwt를 활용한 CUD 작업

- 토큰 시간 100분으로 수정
- dto에 time 추가
- 엔티티의 link -> url 변경

leehyunk6310 commented:
dto에는 duration이 존재하고 entity에는 존재하지 않는 부분 어떻게 하실건지 조율해야 할 듯 합니다

leehyunk6310 commented on src/post/post.service.ts:
```diff
@@ -23,15 +50,62 @@ export class PostService {
   findOne(id: number) {
     return this.postRepository.findOne({
       where: { id },
+      relations: ['author'],
     });
   }
```
relation을 추가해야 post.author에 접근할 수 있습니다.
relation이 없으면 접근시 undefined 값이 출력됩니다

vimkim replied:
맞습니다 감사합니다!

vimkim commented on src/post/post.service.ts:
```diff
@@ -23,15 +50,62 @@ export class PostService {
+    console.log(post.author);
+    console.log(author);
+
+    if (post.author.id != author.id) {
+      throw new UnauthorizedException('Not the owner of Post');
+    }
+
```
비교는 무조건 !==, === 를 사용하는 것이 권장되는 구글, 에어비엔비? 가이드라인입니다.

!== 로 수정바랍니다!

제가 미리 이런 착오가 안 생기시도록 eslint에서 https://eslint.org/docs/latest/rules/eqeqeq 켜두고 구글 가이드라인 켜둬야 했는데 착각했네요 ㅠㅜ

leehyunk6310 replied:
수정하겠습니다 !

vimkim commented on src/post/post.dto.ts:
```diff
@@ -36,6 +36,9 @@ export class PostCreateDto {
   @IsString()
   procedure!: string;
 
+  @IsString()
+  time!: string;
+
```
dto에서 time -> duration으로 rename하기로 어제 현재님과 디코 채팅에서 얘기했었는데 아마 time은 그래서 필요 없을 듯 해요~

vimkim commented:
아이고 entity에 duration이 없는건 실수네요 ㅠㅠ 고쳐주시면 감사드려요

leehyunk6310 committed:
fix: dto, entity 수정

- dto에서 time 제거
- entity에서 time -> duration으로 변경
- != --> !== 로 수정
- jwt 시간 10분으로 수정
